### ISSUE TITLE
Install help?

### ISSUE THREAD

**Issue by Exr0n** [tecosaur/LaTeX-auto-activating-snippets#6]
Hi! Emacs newbie here, this package looks amazing! I can't seem to install it though...

I'm trying to use `laas` with org mode and after manually `(load laas.el)`ing, I have 
```lisp
(use-package aas
  :hook (LaTeX-mode . auto-activating-snippets-mode)
  :hook (org-mode   . auto-activating-snippets-mode)

  :config (require 'laas))

(use-package laas)
```
Which is mostly copied from #1. 

Do I also need to install/setup yasnippet or other dependencies for the smart fraction?

Any suggestions would be greatly appreciated. Thanks!

**Comment by ymarco** [comment]
You really only need

(use-package laas
  :hook (LaTeX-mode . laas-mode)
  ;; if you want it in org-mode too
  :hook (org-mode . laas-mode))

laas should take care of loading aas and activating aas-mode.

> Which is mostly copied from #1.
The info there is outdated. laas used to apply its snippets to LaTeX mode directly which is why only requiring it was sufficient. A few weeks ago I made it its own minor mode.

> Do I have to configure my own snippets?
No, it's not required.

> Can I load the entire default set cleanly?
My code snippets does that.

> And should I be using `latex-auto-activating-snippets` or `laas`?
laas. Sorry for the confusion. The package was renamed to comply with MELPA guidelines.


--
Yoav Marco

**Comment by ymarco** [comment]
> Do I also need to install/setup yasnippet or other dependencies for the smart fraction?

Your package manager should take care of installing the dependencies. No manual configuration of them is required for this package to work.

**Comment by tecosaur** [comment]
It's yet to be added to melpa (a PR is currently open), so a `straight.el` git recipe would currently be the best way IMO.

**Comment by ymarco** [comment]
> The package is mostly working, but I can't seem to get the smart fractions to work.
How exactly do they not work? Emit an error? don't expand at all?
In a LaTeX buffer, when you write $$ and fill in $1/$, it should expand to $\frac{1}{}$.
> I didn't find `laas` on melpa-stable, so how should I be installing it?
Yeah, the PR is still in progress. Expect more minor breakages until the next release.

Meanwhile, I don't know how to install packages from general sources in vanilla Emacs. I know how to do it in Straight and Doom, but... You can install the dependencies {yasnippet, auctex, aas} from MELPA and clone this repo to somewhere in your load-path I guess.

**Comment by Exr0n** [comment]
Thanks for the help!

> >  The package is mostly working, but I can't seem to get the smart fractions to work.
> How exactly do they not work? Emit an error? don't expand at all?
> In a LaTeX buffer, when you write $$ and fill in $1/$, it should expand to $\frac{1}{}$.

This doesn't do anything in org inline math mode, and running `M-x laas-smart-fraction` shows `Wrong type argument: integer-or-marker-p, nil`. However, I just noticed that this works in org full math mode (`\[ .. \]`)? I guess it's an org support problem.

> Meanwhile, I don't know how to install packages from general sources in vanilla Emacs. I know how to do it in Straight and Doom, but... You can install the dependencies {yasnippet, auctex, aas} from MELPA and clone this repo to somewhere in your load-path I guess.

I cloned `laas.el` from this repo and source it in my config, so I guess this works.

> It's yet to be added to melpa (a PR is currently open), so a `straight.el` git recipe would currently be the best way IMO.

I don't know what a `straight.el` git recipe is but it seems to work so I'm a happy camper :)

I love how active this project is, by the way. Thank you all so much!

**Comment by ymarco** [comment]
Exr0n <notifications@github.com> writes:

> > >  The package is mostly working, but I can't seem to get the smart fractions to work.
> > How exactly do they not work? Emit an error? don't expand at all?
> > In a LaTeX buffer, when you write $$ and fill in $1/$, it should expand to $\frac{1}{}$.
> This doesn't do anything in org inline math mode, and running `M-x laas-smart-fraction` shows `Wrong type argument: integer-or-marker-p, nil`.
You're not supposed to call it manually; all snippet-expanding functions are interactive to allow people to put interactive codes in there. Seeing the function in the M-x menu is a byproduct that I'm not sure what to do about.

> However, I just noticed that this works in org full math mode (`\[ .. \]`)? I guess it's an org support problem.
Seems like it the condition, which is laas-object-on-left-condition in the frac snippet's case, doesn't activate. PRs welcome!

**Comment by ymarco** [comment]
>> It's yet to be added to melpa (a PR is currently open), so a `straight.el` git recipe would currently be the best way IMO.

> I don't know what a `straight.el` git recipe is but it seems to work so I'm a happy camper :)

The MELPA PR is merged now, FYI :)

**Comment by Exr0n** [comment]
> > However, I just noticed that this works in org full math mode (`\[ .. \]`)? I guess it's an org support problem.
> 
> Seems like it the condition, which is laas-object-on-left-condition in the frac snippet's case, doesn't activate. PRs welcome!

It looks like `laas-org-mathp` relies on `org-element-type (org-element-context element)` for checking whether an element is a latex context (`\[ \]` not `#+BEGIN_LATEX`) but it returns "paragraph" when in a `$...$` block. I can't find any org apis which give this information, so I can't think of a way of checking this condition unless we see if the line contains a dollar sign before it. And I don't know how to do that efficiently in elisp, so any pointers would be appreciated.

I also found some other unexpected (to me) behavior and will open separate issues about them :{

**Comment by tecosaur** [comment]
Re: Org and `$ ... $` syntax, from the manual:
> Text within the usual LaTeX math delimiters. To avoid conflicts with currency specifications, single `$` characters are only recognized as math delimiters if the enclosed text contains at most two line breaks, is directly attached to the `$` characters with no whitespace in between, and if the closing $ is followed by whitespace, punctuation or a dash. For the other delimiters, there is no such restriction, so when in doubt, use `\(...\)` as inline math delimiters.

https://orgmode.org/manual/LaTeX-fragments.html

TLDR; Just use `\( ... \)`

**Comment by Exr0n** [comment]
Great, this resolves all my petty grievances. I'm loving the plugin, thank you all so much for the development and active support!

For those digging through issues in search of solutions, I solved my math block annoyances with auto-expanding YAS snippets for `\n\\[\n$1\n\\]\n$0` and `\($1\)$0`.

Cheers!
